test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsTestUtils.cs

```diff
@@ -37,6 +38,34 @@ public static string RemoveWhitespace(string s)
             return Regex.Replace(s, @"\s+", string.Empty);
         }
 
+        public static bool SequenceDeepEquals<T>(IList<T> settings1, IList<T> settings2) where T : SettingBase
```

Comment:
Given that this method only inspects and does not mutate its arguments, can you use `IReadOnlyList<T>` instead of `IList<T>`?